kv/memberlist/memberlist_client.go

```diff
@@ -463,15 +465,27 @@ func (m *KV) starting(_ context.Context) error {
 	}
 	m.initWG.Done()
 
+	if m.cfg.JoinInStarting && len(m.cfg.JoinMembers) > 0 {
+		// Lookup SRV records for given addresses to discover members.
+		members := m.discoverMembers(ctx, m.cfg.JoinMembers)
```

Comment:
I would change a couple of things:
1. Remove the new config option `m.cfg.JoinInStarting` (let's try to shrink the config options, not increase them)
2. Only join a subset of nodes in `starting()` and then join all nodes in `running()`
3. Move the call to `discoverMembers()` inside `joinMembersOnStartup()` and call it for each retry. Why? Because if `discoverMembers()` returns a node which is terminating, the join and all retries will fail, while it shouldn't

Reply:
Latest version of PR implements these suggestions.